pkg/server/plugin/datastore/sql/sql.go

```diff
@@ -499,6 +501,22 @@ func (ds *SQLPlugin) withTx(ctx context.Context, op func(tx *gorm.DB) error, rea
 	return sqlError.Wrap(tx.Commit().Error)
 }
 
+// gormToGRPCStatus takes an error, and converts it to a GRPC error.
+// If the error is a gorm error type with a known mapping to a GRPC
+// status, that code will be set, otherwise the code will be set to
+// Unknown.
+func (ds *SQLPlugin) gormToGRPCStatus(err error) error {
```

Comment:
nit: consider making this a standalone helper function since it requires no state from `ds`